src/ProjectService.Data/FileRepository.cs

```diff
@@ -30,11 +32,27 @@ public async Task<List<Guid>> CreateAsync(List<DbProjectFile> files)
       await _provider.SaveAsync();
 
       return files.Select(x => x.FileId).ToList();
+    }
+
+    public async Task<(List<DbProjectFile>, int filesCount)> FindAsync(Guid projectId, BaseFindFilter filter, FileAccessType access = FileAccessType.Manager)
```

Comment:
why do you not use FindProjectFilesFilter?

Reply:
done